sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/HttpPipelineHelper.cs

```diff
@@ -0,0 +1,81 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading;
+using Azure.Core;
+using Azure.Monitor.OpenTelemetry.Exporter.Models;
+
+namespace Azure.Monitor.OpenTelemetry.Exporter
+{
+    internal static class HttpPipelineHelper
+    {
+        private const string RetryAfterHeaderName = "Retry-After";
+
+        internal const int MinimumRetryInterval = 60000;
+
+        internal static byte[] GetRequestContent(RequestContent content)
+        {
+            using MemoryStream st = new MemoryStream();
+
+            content.WriteTo(st, CancellationToken.None);
+
+            return st.ToArray();
+        }
+
+        internal static int GetRetryInterval(HttpMessage message)
+        {
+            if (message.Response.Headers.TryGetValue(RetryAfterHeaderName, out var retryAfterValue))
+            {
+                if (int.TryParse(retryAfterValue, out var delaySeconds))
+                {
+                    return (int)TimeSpan.FromSeconds(delaySeconds).TotalMilliseconds;
+                }
+                if (DateTimeOffset.TryParse(retryAfterValue, out DateTimeOffset delayTime))
+                {
+                    return (int)(delayTime - DateTimeOffset.Now).TotalMilliseconds;
+                }
+            }
+
+            return MinimumRetryInterval;
+        }
+
+        internal static string GetPartialContentFromBreeze(TrackResponse response, string content)
```

Comment:
This method name is misleading, breeze just sends the index of items that are failed with response codes. Please correct me, if my understanding is incorrect.

Reply:
You are right - this method is creating the partial content to retry based on indexes received by breeze. WIll think of better name.